### PR TITLE
add batch api v2 (WP-881)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

**Review by dimitrystd — APPROVED** [review]

File: src/Batch/BatchApiV2.php [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+use Psr\Log\LoggerInterface;
+use Smartling\AuthApi\AuthApiInterface;
+use Smartling\BaseApiAbstract;
+use Smartling\Batch\Params\ListBatchesParameters;
+use Smartling\Exceptions\SmartlingApiException;
+
+class BatchApiV2 extends BaseApiAbstract
+{
+    public const ENDPOINT_URL = 'https://api.smartling.com/jobs-batches-api/v2/projects';
+    private const GET_BATCH_STATUS_ARRAY_SHAPE = [
```
Comment (PavelLoparev) [inline src/Batch/BatchApiV2.php]: This is nonsense. Typehinting through the php doc.. It makes no sense for type safety at all. Moreover, works only in php storm. why should I read these lines in another ide?

File: src/Batch/BatchApiV2.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            self::HTTP_METHOD_GET,
+        );
+    }
+
+    /**
+     * @throws SmartlingApiException
+     */
+    public function processBatchAction(
+        string $batchUid,
+        #[ExpectedValues(['CANCEL_FILE', 'REGISTER_FILE'])]
```
Comment (PavelLoparev) [inline src/Batch/BatchApiV2.php]: The same. It's only for autocompletion, it's ridiculous. You can put whatever you want in `$action` - nothing happens.

File: src/Batch/BatchApiV2.php [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    /**
+     * @throws SmartlingApiException
+     */
+    #[ArrayShape(['batchUid' => 'string'])]
+    public function createBatch(bool $authorize,
+        string $translationJobUid,
+        array $fileUris,
+        array $localeWorkflows = []
```
Comment (PavelLoparev) [inline src/Batch/BatchApiV2.php]: Why don't we introduce params dto?

File: src/Batch/BatchApiV2.php [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+    public function uploadFileToABatch(
+        string $batchUid,
+        string $file,
+        string $fileUri,
+        string $fileType,
+        array $localeIdsToAuthorize,
+        string $smartlingNamespace = null,
+        string $smartlingFileCharset = null,
+        string $callbackUrl = null
```
Comment (PavelLoparev) [inline src/Batch/BatchApiV2.php]: Same here - looks weird we don't pass params object here as usual

**Review by PavelLoparev — CHANGES_REQUESTED** [review]
You can ignore comments about `ArrayShape` etc phpdoc "helpers" but in general it makes no sense at all. It probably even only works in some exact version of phpstorm, idk

File: src/Batch/BatchApiV2.php (hunk 1 above)
Comment (PavelLoparev) [reply src/Batch/BatchApiV2.php]: I mean, our code "knows" about the ide it's being edited in, this is weird. idk, maybe I don't get the whole idea

**Comment by vsolovei-smartling** [comment]
I've removed unnecessary code

**Review by dimitrystd — APPROVED** [review]

**Review by PavelLoparev — APPROVED** [review]

**Pull request merged**